src/rdkafka_conf.c

```diff
@@ -3284,6 +3302,26 @@ const char *rd_kafka_conf_finalize (rd_kafka_type_t cltype,
                 conf->eos.idempotence = 0;
 
         } else if (cltype == RD_KAFKA_PRODUCER) {
+                if (conf->eos.transactional_id) {
+                        if (!conf->eos.idempotence) {
+                                /* Auto enable idempotence unless
+                                 * explicitly disabled */
+                                if (rd_kafka_conf_is_modified(
+                                            conf, "enable.idempotence"))
+                                        return "`transactional.id` requires "
+                                                "`enable.idempotence=true`";
+
+                                conf->eos.idempotence = rd_true;
+                        }
+
+                        /* Make sure at least one request can be sent
+                         * before the transaction times out. */
+                        if (conf->eos.transaction_timeout_ms <
+                            conf->socket_timeout_ms)
+                                return "`transaction.timeout.ms` must be "
```

Comment:
does `transaction.timeout.ms` need to be greater than equal to `socket.timeout.ms` or greater than equal to `socket.timeout.ms` + 1000

>The transaction timeout must be at least 1000 ms larger than `message.timeout.ms` and `socket.timeout.ms`

Reply:
Good catch :+1:

To keep things simple for the user we probably want to make transaction.timeout.ms the defining property, automatically adjusting socket.timeout.ms (unless explicitly configured < transaction.timeout.ms) and message.timeout.ms (set to transaction.timeout.ms).